libcaf_core/caf/detail/format.hpp

```diff
@@ -22,21 +25,59 @@
 
 namespace caf::detail {
 
+template <class T>
+inline constexpr bool is_formattable = is_complete<std::formatter<T, char>>;
+
+template <class T>
+inline constexpr bool fmt_needs_to_string
+  = !is_formattable<T> && has_inspect_overload_v<stringification_inspector, T>;
+
+template <class T>
+constexpr T&&
+fmt_fwd(std::remove_reference_t<T>& arg,
+        std::enable_if_t<is_formattable<std::decay_t<T>>>* = nullptr) noexcept {
+  return static_cast<T&&>(arg);
```

Comment:
Isn't this unsafe? You cast an lvalue reverence to an rvalue reference?

Reply:
No, that's pretty much the canonical implementation of `forward`.

Say you have the expression, `type&&`. If `type` is `foo`, this will deduce to `foo&&`. However, if `type` is `foo&`, it will deduce to just `foo&`. This is called "reference collapsing" (e.g., see https://en.cppreference.com/w/cpp/language/reference).

Reply:
Oops, I forgot that template references work that way. Thanks for explaining it.